nominal/nominal.py

```diff
@@ -1,628 +1,316 @@
 from __future__ import annotations
 
-import copy
-import io
-import os
-from datetime import datetime, timedelta
-from math import floor
+import shutil
+from datetime import datetime
 from pathlib import Path
-from typing import Mapping, Optional, Sequence
-
-import jsondiff as jd
-import keyring as kr
-import polars as pl
-import requests
-from dateutil import parser
-from jsondiff import diff
-from rich import print
-
-from ._api.combined.ingest_api import (
-    AbsoluteTimestamp,
-    IngestService,
-    IngestSource,
-    RelativeTimestamp,
-    S3IngestSource,
-    TimestampMetadata,
-    Iso8601Timestamp,
-    TimestampType,
-    TriggerIngest,
+from threading import Thread
+from typing import TYPE_CHECKING, BinaryIO, Mapping, Sequence
+
+import dateutil.parser
+
+from .exceptions import NominalError
+from .sdk import Attachment, Dataset, NominalClient, Run
+from ._utils import (
+    FileType,
+    FileTypes,
+    IntegralNanosecondsUTC,
+    TimestampColumnType,
+    _datetime_to_integral_nanoseconds,
+    reader_writer,
 )
-from ._utils import PayloadFactory, create_service, default_filename
-from .cloud import ENDPOINTS, _auth_help_blurb, get_app_base_url, get_base_url
 
+if TYPE_CHECKING:
+    import pandas as pd
+    import polars as pl
```

Comment:
interesting, when is this `TYPE_CHECKING` var defined? is this is mypy thing

Reply:
Yeah. If users aren't using pandas, they don't pay an import penalty for pandas/polars just because I included it in a type hint

Reply:
🔥